petab/core.py

```diff
@@ -770,3 +770,66 @@ def measurements_have_replicates(measurement_df: pd.DataFrame) -> bool:
             measurement_df,
             ['observableId', 'simulationConditionId',
              'preequilibrationConditionId', 'time'])).size().values - 1)
+
+
+def flatten_timepoint_specific_output_overrides(
+        petab_problem: Problem) -> None:
+    """If the PEtab problem definition has timepoint-specific
+    observableParameters or noiseParameters for the same observable, replace
+    those by replicating the respective observable.
+
+    This is a helper function for some tools which may not support such
+    timepoint-specific mappings.
+
+    Arguments:
+        petab_problem:
+            PEtab problem to work on
+    """
+
+    df = petab_problem.measurement_df[["observableId", "preequilibrationConditionId", "simulationConditionId"]]
+    df_unique_values = df.drop_duplicates()
+    df_new = pd.DataFrame()
+    for nrow in range(len(df_unique_values.index)):
+        df = petab_problem.measurement_df.loc[(petab_problem.measurement_df['observableId'] >= df_unique_values.iloc[nrow, :].values[0])
+                                         & (petab_problem.measurement_df['preequilibrationConditionId'] <= df_unique_values.iloc[nrow, :].values[1])
```

Comment:
The `values[n]` are a bit tough to read. Can you use string-based indexing?

Reply:
done